keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Resource.kt

```diff
@@ -41,10 +44,15 @@ data class Resource<T : ResourceSpec>(
 /**
  * External representation of a resource that would be submitted to the API
  */
-data class SubmittedResource<T : Any>(
+data class SubmittedResource<T : ResourceSpec>(
```

Comment:
This is the key thing with this PR. Being able to strongly type the spec at the REST boundary so we don't have to do any messing about converting things.